source/ptreferential/ptreferential_ng.cpp

```diff
@@ -412,16 +429,19 @@ std::string make_request(const Type_e requested_type,
             break;
         case Type_e::VehicleJourney:
         case Type_e::Impact:
-            if (since && until) {
-                res = "(" + res + ") AND " + static_data->captionByType(requested_type) + ".between("
-                      + to_iso_string(*since) + "Z, " + to_iso_string(*until) + "Z, " + rt_level_to_string(rt_level)
-                      + ")";
-            } else if (since) {
-                res = "(" + res + ") AND " + static_data->captionByType(requested_type) + ".since("
-                      + to_iso_string(*since) + "Z, " + rt_level_to_string(rt_level) + ")";
-            } else if (until) {
-                res = "(" + res + ") AND " + static_data->captionByType(requested_type) + ".until("
-                      + to_iso_string(*until) + "Z, " + rt_level_to_string(rt_level) + ")";
+            if (since || until) {
+                res = "(" + res + ") AND " + static_data->captionByType(requested_type);
+                if (since && until) {
+                    res = res + ".between(" + to_iso_string(*since) + "Z, " + to_iso_string(*until) + "Z";
```

Comment:
`res += ...`

Reply:
Then the "blabla" is a const char*, no operator+ available, needs a cast.
So it's shorter and more readable currently to me :neutral_face: